src/watchdog/utils/patterns.py

```diff
@@ -0,0 +1,75 @@
+#!/usr/bin/env python
+# -*- coding: utf-8 -*-
+# patterns.py: Common wildcard searching/filtering functionality for files.
+#
+# Copyright (C) 2010 Yesudeep Mangalapilly <yesudeep@gmail.com>
+#
+# Written by Boris Staletic <boris.staletic@gmail.com>
+
+from pathlib import PureWindowsPath, PurePosixPath
```

Comment:
I would like a comment explaining the magic here :)
- Why doing that import?
- Why using them later instead of simple `Path`. And note that it relies on the CPython implementation that ensure the case-sensitivity constraint. Maybe by link to https://docs.python.org/3/library/pathlib.html#pathlib.PurePath.match.